code/modules/antagonists/mind_flayer/powers/flayer_stealth_powers.dm

```diff
@@ -0,0 +1,256 @@
+/// Hack computer cameras to use them as a secret camera network
+/datum/spell/flayer/surveillance_monitor
+	name = "Camfecting Bug"
+	desc = "Allows us to cast a hack to a computers webcam. Alt-click the spell to access all your hacked computer webcams."
+	power_type = FLAYER_PURCHASABLE_POWER
+	category = FLAYER_CATEGORY_INTRUDER
+	base_cooldown = 1 SECONDS
+	action_icon = 'icons/obj/device.dmi'
+	action_icon_state = "camera_bug"
+	base_cost = 50
+	static_upgrade_increase = 15
+	stage = 1
+	max_level = 4
+	upgrade_info = "Upgrades increase the amount of computers you can hack by 6."
+	/// An internal camera bug
+	var/obj/item/camera_bug/internal_camera
+	/// How many computers can we have hacked at most?
+	var/maximum_hacked_computers = 6
+	/// List of references to the bugs inside the computers that we hacked
+	var/list/active_bugs = list()
+
+/datum/spell/flayer/surveillance_monitor/Destroy(force, ...)
+	. = ..()
+	active_bugs = null
```

Comment:
You new the camera bugs on alt_click, will they get deleted? I think you might want to QDEL_LIST_CONTENTS here, or at the very least do `active_bugs.Cut()` instead of just nulling it out.

Might also want to do `if(!QDELETED())` on internal_camera, otherwise it might just get stuck to things

Reply:
Done!